nncf/experimental/torch/fx/nncf_graph_builder.py

```diff
@@ -0,0 +1,143 @@
+        :return: NNCFGraph.
+        """
+
+        nncf_graph = PTNNCFGraph()
+
+        for source_node in model.graph.nodes:
+            node_type, node_metatype = GraphConverter._get_node_type_and_metatype(source_node)
+
+            nncf_graph.add_nncf_node(
+                node_name=source_node.name,
+                node_type=node_type,
+                node_metatype=node_metatype,
+            )
+
+        for source_node in model.graph.nodes:
+            source_nncf_node = nncf_graph.get_node_by_name(source_node.name)
+            for idx, dist_node in enumerate(source_node.users):
+                dist_node_id = nncf_graph.get_node_by_name(dist_node.name).node_id
+                input_port_id, output_port_id, tensor_shape = GraphConverter.get_edge_params(
+                    model, source_node, source_nncf_node, dist_node, idx
+                )
+
+                nncf_graph.add_edge_between_nncf_nodes(
+                    source_nncf_node.node_id,
+                    dist_node_id,
+                    tensor_shape=tensor_shape,
+                    input_port_id=input_port_id,
+                    output_port_id=output_port_id,
+                    dtype=Dtype.FLOAT,
+                )
+
+        return nncf_graph
+
+    @staticmethod
+    def get_edge_params(
+        model: torch.fx.GraphModule,
+        source_node: torch.fx.Node,
+        source_nncf_node: NNCFNode,
+        dist_node: torch.fx.Node,
+        output_idx: int,
+    ) -> Tuple[int, int, Tuple[int, ...]]:
+        """
+        Retrieves edge params from the given source_node and dist_node pair.
+
+        :param model: A torch.fx.GraphModule instance.
+        :param source_node: Source node in format of torch.fx.Node.
+        :param source_nncf_node: Source node in format of NNCFNode.
+        :param dist_node: Distance node in format of torch.fx.Node.
+        :param output_idx: Output indes of the source_node.
+        :return: Tuple of edge parameters: edge input port id, edge output port id and
+            edge tensor shape.
+        """
+        output_port_id = 0
+        if source_node.op in ("get_attr",):
+            tensor = getattr(model, source_node.target)
+            tensor_shape = tuple(tensor.shape)
+        elif "val" in source_node.meta:
+            if source_nncf_node.metatype is om.PTBatchNormMetatype:
+                tensor = source_node.meta["val"][0]
+            elif source_nncf_node.metatype is om.PTSplitMetatype:
+                tensor = source_node.meta["val"][output_idx]
+                # Assume every split outputs corresponds to an unique output_port_id
+                output_port_id = output_idx
+            else:
+                tensor = source_node.meta["val"]
+            tensor_shape = tuple(tensor.shape)
+        else:
+            nncf_logger.info(f"Edge shape between {source_node.name} and {dist_node.name} is unknown.")
+            tensor_shape = None
+        if "quantize" in dist_node.name:
```

Comment:
This is the addition for adding shape to quantize, dequantize node edges. It essentially checks if the next node is a quantize or dequantize node and adds the value of current node to that node so the edge is added when traversing the qdq node.

Reply:
I don't think we should set any values inside the function which is reading values, I believe we need to find a way this meta is being build and rebuild it automatically using the torch fx functionality

Reply:
yeah, that sounds right. I was finding a way to add it into the qdq function for torch fx.